Code/Framework/AzToolsFramework/AzToolsFramework/ComponentMode/ComponentModeDelegate.h

```diff
@@ -91,20 +94,23 @@ namespace AzToolsFramework
                 EditorComponentSelectionRequestsBus::Handler* handler);
 
             /// Utility function for the common case of creating a single ComponentMode for a Component.
-            template<typename EditorComponentType, typename EditorComponentModeType>
-            void IndividualComponentMode()
+            template<typename EditorComponentType, typename EditorComponentModeType, typename ...Params>
+            void IndividualComponentMode(Params&&... params)
             {
-                SetAddComponentModeCallback([](const AZ::EntityComponentIdPair& entityComponentIdPair)
-                {
-                    const auto componentModeBuilder =
-                        CreateComponentModeBuilder<EditorComponentType, EditorComponentModeType>(entityComponentIdPair);
-
-                    const auto entityAndComponentModeBuilder =
-                        EntityAndComponentModeBuilders(entityComponentIdPair.GetEntityId(), componentModeBuilder);
-
-                    ComponentModeSystemRequestBus::Broadcast(
-                        &ComponentModeSystemRequests::AddComponentModes, entityAndComponentModeBuilder);
-                });
+                SetAddComponentModeCallback(AZStd::bind(
+                    [](const AZ::EntityComponentIdPair& entityComponentIdPair, Params&&... params)
+                    {
+                        const auto componentModeBuilder = CreateComponentModeBuilder<EditorComponentType, EditorComponentModeType>(
+                            entityComponentIdPair, AZStd::forward<Params>(params)...);
+
+                        const auto entityAndComponentModeBuilder =
+                            EntityAndComponentModeBuilders(entityComponentIdPair.GetEntityId(), componentModeBuilder);
+
+                        ComponentModeSystemRequestBus::Broadcast(
+                            &ComponentModeSystemRequests::AddComponentModes, entityAndComponentModeBuilder);
+                    },
+                    AZStd::placeholders::_1,
+                    AZStd::forward<Params>(params)...));
```

Comment:
nit: It's generally better to just use a lambda these days and totally avoid `AZStd::bind` as it's a library feature that can now be achieved in the language (using lamdas) and `AZStd::bind` is super slow to compile

Reply:
I did it that way because I read something about it being difficult to capture variadic arguments and using bind was the workaround the article suggested.

Reply:
Ah okay makes sense, thanks for clarifying